### ISSUE TITLE
Oauth warning over https with peertube using nginx server external to docker 

### ISSUE THREAD

**Issue by mhfowler** [Chocobozzz/PeerTube#4087]
Everything appears to be working well over http. Everything also appears to be working well over https, except when you first arrive at the page I see a modal with the warning:
``` basj
Cannot retrieve OAuth Client credentials: undefined. Ensure you have correctly configured PeerTube (config/ directory), in particular the "webserver" section.
```

I'm not sure what I've configured incorrectly. I'm using a nginx server external to docker, with peertube running within docker based on the standard installation. 

Here are my relevant configurations:

This is my nginx config, which is outside of docker, in /etc/nginx/sites-enabled/videos.hyphalfusion.network:
``` bash
server {
	listen 80;
        server_name videos.hyphalfusion.network;

        location / {
           proxy_pass http://127.0.0.1:9000;
           proxy_set_header   Host $host;
           proxy_set_header   X-Real-IP $remote_addr;
           proxy_set_header   X-Forwarded-For $proxy_add_x_forwarded_for;
           proxy_set_header   X-Forwarded-Host $server_name;
        }
}

server {
    listen 443 ssl;
    server_name videos.hyphalfusion.network;

    # ssl configuration;
    ssl_certificate /etc/letsencrypt/live/hyphalfusion.network/fullchain.pem;
    ssl_certificate_key /etc/letsencrypt/live/hyphalfusion.network/privkey.pem;

    location / {
                proxy_pass http://127.0.0.1:9000;
		proxy_set_header X-Forwarded-Proto $scheme;
           	proxy_set_header   X-Forwarded-Host $server_name;
           	proxy_set_header   X-Real-IP $remote_addr;
                proxy_read_timeout      90;
                proxy_redirect  http://videos.hyphalfusion.network:9000/ https://videos.hyphalfusion.network;
    }
}
```

These are the lines I think are relevant in docker-compose.yml. I've commented out the certbot and webserver sections.
``` bash
  peertube:
    # If you don't want to use the official image and build one from sources:
    # build:
    #   context: .
    #   dockerfile: ./support/docker/production/Dockerfile.buster
    image: chocobozzz/peertube:production-buster
    # Use a static IP for this container because nginx does not handle proxy host change without reload
    # This container could be restarted on crash or until the postgresql database is ready for connection
    networks:
      default:
        ipv4_address: 172.18.0.43
    env_file:
      - .env

    ports:
     - "1935:1935" # If you don't want to use the live feature, you can comment this line
     - "9000:9000" # If you provide your own webserver and reverse-proxy, otherwise not suitable for production
    volumes:
      - assets:/app/client/dist
      - ./docker-volume/data:/data
      - ./docker-volume/config:/config
    depends_on:
      - postgres
      - redis
      - postfix
    restart: "always"
```

And here are lines from .env
``` bash
# Server configuration
PEERTUBE_WEBSERVER_HOSTNAME=videos.hyphalfusion.network
# If you do not use https and a reverse-proxy in docker-compose.yml
PEERTUBE_WEBSERVER_PORT=443
PEERTUBE_WEBSERVER_HTTPS=true
# If you need more than one IP as trust_proxy
# pass them as a comma separated array:
PEERTUBE_TRUST_PROXY=["127.0.0.1", "loopback", "172.18.0.0/16"]
```

any ideas?

**Comment by rigelk** [comment]
Why are you using `proxy_redirect  http://videos.hyphalfusion.network:9000/ https://videos.hyphalfusion.network;` instead of [our https redirection](https://github.com/Chocobozzz/PeerTube/blob/develop/support/nginx/peertube#L15)?

**Comment by mhfowler** [comment]
@rigelk I just didn't realize that was already happening. I've tried uncommenting this line and still see the oauth error though. 

Is there an example of an nginx config I should use to proxy to peertube?

**Comment by Chocobozzz** [comment]
If you follow our [docker guide](https://docs.joinpeertube.org/install-docker?id=install), you should already have a working nginx config in `docker-volume`. I suggest you to reinstall peertube using this guide

**Comment by mhfowler** [comment]
thanks @Chocobozzz. 

I needed to not use the nginx as a docker-container, because I was running peertube on a machine that already was using nginx to serve some other apps too. So ports 80 and 443 were already in use. 

But using [the nginx config](https://raw.githubusercontent.com/Chocobozzz/PeerTube/develop/support/nginx/peertube) you mentioned as the nginx config for my nginx running external to docker worked for me. I just substituted ${WEBSERVER_HOST} with my domain, and ${PEERTUBE_HOST} with `127.0.0.1:9000`

Initially I then also saw a blank screen when I loaded the site (static resources had failed to load). 

I then commented out all the sections of the nginx config that said "bypass peertube" optional, restarted nginx, and then everything was working with no error message. Just writing this out in case its helpful to anyone else. It might be clear to others, but for me reading the guide, it wasn't clear what you were supposed to do if you needed to run nginx outside of the docker. Or maybe there is even a way to run nginx inside of docker without the port clashing but I wasn't sure.
